src/state.ts

```diff
@@ -11,37 +11,45 @@ import * as vscode from 'vscode';
  * invalid states.
  */
 export class StateManager {
-  constructor(readonly extensionContext: vscode.ExtensionContext) {}
+  constructor(readonly extensionContext: vscode.ExtensionContext, readonly folder: vscode.WorkspaceFolder) {}
+
+  private _get<T>(key: string): T | undefined {
+    return this.extensionContext.globalState.get<T>(this.folder.uri.fsPath + key);
```

Comment:
The main issue is when there's an existing single root project and a new folder is added. Let's say user has been working on project A, and it has kits set. Then folder B is added. Now since the workspace is changed, the user will be prompted to select kit again.

Reply:
That makes sense. But I think that is the exception rather than the rule. You could have some global state that tracks the most recently active kit which is used when there is no value for the workspaceState yet (because it is a new workspace).  I think it is reasonable to require the user to pick the target and maybe the variant again when the workspace adds a second folder, but you could follow this fallback pattern for all properties if you like.